src/Framework/AspNetCoreAnalyzers/test/WebApplicationBuilder/UseTopLevelRouteRegistrationInsteadOfUseEndpointsTest.cs

```diff
@@ -58,6 +58,48 @@ public static void Main (string[] args)
         Assert.Empty(diagnostics);
     }
 
+    [Fact]
+    public async Task DoesNotWarnWhenEndpointRegistrationIsNotTopLevel_NoInvocation()
+    {
+        //arrange
+        var source = TestSource.Read(@"
+using Microsoft.AspNetCore.Builder;
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+app.UseRouting();
+app./*MM*/UseEndpoints(e => { });
```

Comment:
```suggestion
app.UseEndpoints(e => { });
```

The diagnostic marker in the source code is not necessary since we're not checking for the location of a particular diagnostic.